Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -161,97 +163,79 @@ public void Serialize<T>(Stream stream, T item)
 
         public void Serialize<T>(IIonWriter writer, T item)
         {
+            currentDepth++;
+            if (currentDepth >= options.MaxDepth)
+            {
+                currentDepth = 0;

```

Comment:
It struck me as odd to see this reset to 0 before throwing an exception, would it be clearer to use a try/finally where you decrement in the finally block and not have this 0 set in both Serialize/Deserialize case?

Reply:
Now using try-catch